src/sst/core/elemLoader.cc

```diff
@@ -135,92 +73,114 @@ splitPath(const std::string& searchPaths)
-            "Opening and resolving references for element library %s failed:\n"
-            "\t%s\n",
-            elemlib.c_str(), dlerror());
-        err_os << (const char*)err_str.data();
+    else if ( (!strcmp(bind_env, "now")) || (!strcmp(bind_env, "NOW")) ) {
+        bindPolicy = RTLD_NOW | RTLD_GLOBAL;
     }
 }
 
+ElemLoader::~ElemLoader() {}
+
 void
 ElemLoader::loadLibrary(const std::string& elemlib, std::ostream& err_os)
 {
-    std::string libname = "lib" + elemlib;
-    lt_dlhandle lt_handle;
-    lt_handle = lt_dlopenadvise(libname.c_str(), loaderData->advise_handle);
-    if ( nullptr == lt_handle ) {
-        // The preopen module runs last and if the
-        // component was found earlier, but has a missing symbol or
-        // the like, we just get an amorphous "file not found" error,
-        // which is totally useless...
-        // Make darn sure we have enough space to hold the error message
-        std::vector<char> err_str(1e6);
-        sprintf(err_str.data(), "Opening element library %s failed: %s\n", elemlib.c_str(), lt_dlerror());
-        err_os << (const char*)err_str.data();
-        followError(libname, elemlib, searchPaths, err_os);
-    }
+    std::vector<std::string> paths   = splitPath(searchPaths);
+
+    char* full_path     = new char[PATH_MAX];
+    bool  found_element = false;
 
-    // loading a library can "wipe" previously loaded libraries depending
-    // on how weak symbol resolution works in dlopen
-    // rerun the loaders to make sure everything is still registered
-    for ( auto& libpair : ELI::LoadedLibraries::getLoaders() ) {
-        // loop all the elements in the element lib
-        for ( auto& elempair : libpair.second ) {
-            // loop all the loaders in the element
-            for ( auto* loader : elempair.second ) {
-                loader->load();
+    for ( std::string& next_path : paths ) {
+        if ( verbose ) { printf("SST-DL: Searching: %s\n", next_path.c_str()); }
+
+        if ( next_path.at(next_path.size() - 1) == '/' ) {
+            sprintf(full_path, "%slib%s.so", next_path.c_str(), elemlib.c_str());
+        }
+        else {
+            sprintf(full_path, "%s/lib%s.so", next_path.c_str(), elemlib.c_str());
+        }
+
+        if ( verbose ) { printf("SST-DL: Attempting to load %s\n", full_path); }
+
+        // use a global bind policy read from environment, default to RTLD_LAZY
+        void* handle = dlopen(full_path, bindPolicy);
+
+        if ( nullptr == handle ) {
+            if ( verbose ) { printf("SST-DL: Loading failed, error: %s\n", dlerror()); }
+        }
+        else {
+            if ( verbose ) { printf("SST-DL: Load was successful.\n"); }
+
+            found_element = true;
+
+            for ( auto& libpair : ELI::LoadedLibraries::getLoaders() ) {
+                // loop all the elements in the element lib
+                for ( auto& elempair : libpair.second ) {
+                    // loop all the loaders in the element
```

Comment:
Comment probably is redundant IMO, I think a for each loop with a variable name loader is self explanatory.  Same with the above about looping the elements in the element lib.